diff --git a/include/gazebo_aruco_plugin.h b/include/gazebo_aruco_plugin.h
index 9e0249d..57a07c3 100644
--- a/include/gazebo_aruco_plugin.h
+++ b/include/gazebo_aruco_plugin.h
@@ -38,8 +38,15 @@
 #include "TargetRelative.pb.h"
 
 #include <opencv2/opencv.hpp>
+
 #include <opencv2/aruco.hpp>
+// Check for OpenCV version 4.7 or later
+#if CV_VERSION_MAJOR >= 4 && CV_VERSION_MINOR >= 7
+#include <opencv2/objdetect/aruco_dictionary.hpp>
+#else
 #include <opencv2/aruco/dictionary.hpp>
+#endif
+
 #include <opencv2/core.hpp>
 
 #include <vector>
diff --git a/src/gazebo_aruco_plugin.cpp b/src/gazebo_aruco_plugin.cpp
index 27b6b1c..eacacb3 100644
--- a/src/gazebo_aruco_plugin.cpp
+++ b/src/gazebo_aruco_plugin.cpp
@@ -166,8 +166,14 @@ void arucoMarkerPlugin::Load(sensors::SensorPtr _sensor, sdf::ElementPtr _sdf)
   //init marker detection
   this->markerLength_mm = 6.f/8 * 1000 * box_length; 
   int dict_id = 0; 
-  this->marker_dict = cv::aruco::getPredefinedDictionary(cv::aruco::PREDEFINED_DICTIONARY_NAME(dict_id));  
-  this->detectorParams = cv::aruco::DetectorParameters::create();
+  // Check for OpenCV version 4.7 or later
+  #if CV_VERSION_MAJOR >= 4 && CV_VERSION_MINOR >= 7
+    this->marker_dict = makePtr<aruco::Dictionary>(cv::aruco::getPredefinedDictionary(cv::aruco::DICT_4X4_50)); 
+    this->detectorParams = makePtr<aruco::DetectorParameters>(cv::aruco::DetectorParameters());
+  #else
+    this->marker_dict = cv::aruco::getPredefinedDictionary(cv::aruco::PREDEFINED_DICTIONARY_NAME(dict_id));  
+    this->detectorParams = cv::aruco::DetectorParameters::create();
+  #endif
 }
 
 /////////////////////////////////////////////////
